src/openrct2/windows/footpath.c

```diff
@@ -914,6 +937,16 @@ static void window_footpath_construct()
 			gFootpathConstructFromPosition.z
 		);
 
+		// Get path scenery
+		uint16 add_scenery = window_footpath_scenery_get_next_scenery();
+		if (add_scenery != 0)
+		{
+			uint16 sceneryFlags = (add_scenery & 0xFF) + 1;
```

Comment:
No, 0xFF + 1 (worst case) is 0x100, which fits in an uint16. And as stated in the earlier review, this is the code that is used for placing path scenery in top_toolbar.c.